std/format.d

```diff
@@ -4381,7 +4382,9 @@ T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
         spec.spec == 'b' ? 2 :
         spec.spec == 's' || spec.spec == 'd' || spec.spec == 'u' ? 10 : 0;
     assert(base != 0);
-    return parse!T(input, base);
+
+    static if(__traits(compiles, parse!T(input, base)) ) return parse!T(input, base);
```

Comment:
Same issue here - you can't have the function fail to compile. Move the static test to the function constraint.